pkg/actors/actors.go

```diff
@@ -434,7 +436,7 @@ func (a *actorsRuntime) callLocalActor(ctx context.Context, req *invokev1.Invoke
 
 	// The .NET SDK signifies Actor failure via a header instead of a bad response.
 	if _, ok := resp.Headers()["X-Daprerrorresponseheader"]; ok {
-		return nil, errors.Errorf("error from a .NET actor")
+		return resp, ErrDaprResponseHeader
```

Comment:
Just curious, why return `resp` _and_ the error?

Reply:
In the case of .NET actors, they communicate the error via the header in the response. If we don't return the response, the .NET actor runtime gets very upset and is a breaking behavior change.

Reply:
For the record, before I inspected for this header, this request would've returned normally and we wouldn't have seen any error at all. So this is a return to earlier behavior.